File: cpp/design_patterns/constructor/src/EngineDummy.cpp
#include <stdio.h>
#include "EngineDummy.h"

EngineDummy::EngineDummy() {
}


EngineDummy::~EngineDummy() {
}

void EngineDummy::Start() {
    printf("%s: Starting Engine\n", __func__);
}

void EngineDummy::ShutDown() {
    printf("%s: Stopping Engine\n", __func__);
}

File: java/design/contructor/src/contructor/EngineDummy.java
package contructor;

public class EngineDummy implements Engine {

  private Logger logger;

  static EngineDummy GetInstance(Logger logger) {
    return new EngineDummy(logger);
  }
  
  private EngineDummy(Logger logger) {
    this.logger = logger;
  }

  @Override
  public void Start() {
    logger.Log("EngineDummy: Start");
  }

  @Override
  public void Stop() {
    logger.Log("EngineDummy: Stop");
  }

}
